File: ru.kmorozov.gbd.db/src/main/java/ru/kmorozov/gbd/core/logic/model/book/shpl/ShplPagesInfo.java
package ru.kmorozov.gbd.core.logic.model.book.shpl;

import com.google.gson.annotations.SerializedName;
import ru.kmorozov.gbd.core.logic.model.book.base.IPage;
import ru.kmorozov.gbd.core.logic.model.book.base.IPagesInfo;

/**
 * Created by sbt-morozov-kv on 16.11.2016.
 */
public class ShplPagesInfo implements IPagesInfo {

    @SerializedName("page")
    private final ShplPage[] pages;

    public ShplPagesInfo(final ShplPage[] pages) {
        this.pages = pages;
    }

    @Override
    public ShplPage[] getPages() {
        return pages;
    }

    @Override
    public String getMissingPagesList() {
        return null;
    }

    @Override
    public void build() {

    }

    @Override
    public IPage getPageByPid(final String pid) {
        return null;
    }
}

File: ru.kmorozov.gbd.db/src/main/java/ru/kmorozov/db/core/logic/model/book/shpl/ShplPagesInfo.kt
package ru.kmorozov.db.core.logic.model.book.shpl

import com.google.gson.annotations.SerializedName
import ru.kmorozov.db.core.logic.model.book.shpl.ShplPage.Companion.EMPTY_SHPL_PAGE
import ru.kmorozov.gbd.core.logic.model.book.base.AbstractPagesInfo
import ru.kmorozov.gbd.core.logic.model.book.base.IPage

/**
 * Created by sbt-morozov-kv on 16.11.2016.
 */
class ShplPagesInfo(pages: Array<IPage>) : AbstractPagesInfo() {

    @field:SerializedName("page")
    override val pages = pages

    override fun getPageByPid(pid: String): IPage {
        return EMPTY_SHPL_PAGE
    }
}
